dp/generate_climos.py

```diff
@@ -366,10 +370,29 @@ def update_metadata_and_time_var(input_file, t_start, t_end, climo_filepath):
             raise ValueError('Expected climo file to contain # time values in {}, but found {}'
                              .format(num_times_to_interval_set.keys(), cf.time_var.size))
 
+
```

Comment:
You should validate `operation` here too. 

More generally, each function that takes it as an argument needs to do this. It's a bit repetitive, but it catches errors where they occur (e.g., mangling or forgetting to pass an argument). Given that, you probably want to create a utility function `validate_operation` that does this for you, and call it in each place.